### PR TITLE
Release v0.10.0

### PR DESCRIPTION
**Pull request by bfops**
## Description of Changes
This is just the release branch (this should not merge into `master`).

### CODE REVIEW

File: examples/quickstart/client/client.csproj [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net7.0</TargetFramework>
-    <ImplicitUsings>disable</ImplicitUsings>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (RReverser) [inline examples/quickstart/client/client.csproj:5]: Why are these necessary? It should work without those changes (at least it does locally).

File: SpacetimeDB.ClientSDK.csproj [hunk 2]
```diff
@@ -17,8 +17,8 @@
     <PackageIcon>logo.png</PackageIcon>
     <PackageReadmeFile>README.md</PackageReadmeFile>
     <RepositoryUrl>https://github.com/clockworklabs/spacetimedb-csharp-sdk</RepositoryUrl>
-    <AssemblyVersion>0.8.0</AssemblyVersion>
-    <Version>0.9.2</Version>
+    <AssemblyVersion>0.10.0</AssemblyVersion>
+    <Version>0.10.0</Version>
```
Comment (RReverser) [inline SpacetimeDB.ClientSDK.csproj:21]:
```suggestion
    <Version>$(AssemblyVersion)</Version>
```

to make future updates easier

File: examples/quickstart/client/client.csproj (hunk 1 above)
Comment (bfops) [reply examples/quickstart/client/client.csproj:5]: I've seen some of our libs complain about net framework mismatch, no idea if they apply here, happy to revert.

Comment (RReverser) [reply examples/quickstart/client/client.csproj]: Please do, and then we'll land this PR since all the other changes are relevant and needed.

**Review by RReverser — APPROVED** [review]
